### PR TITLE
Update C++ Compilers to clang 15 and gcc 11

### PR DESCRIPTION
**Pull request by sfc-gh-dluan**
Install `devtoolset-11` to update gcc to version 11.2.1 
Compile and install llvm 15.0.6
Bump `boringssl` version - current version does not compile with gcc11.

### CODE REVIEW

File: docker/centos7/Dockerfile [hunk 1]
```diff
@@ -23,11 +23,12 @@ RUN if [ ! "$(uname -m)" == "aarch64" ]; then \
         curl \
         debbuild \
         devtoolset-8 \
```
Comment (ammolitor) [inline docker/centos7/Dockerfile:25]: This line looks, to me, like it should have been replaced by the line below. Is there a reason to keep this package?

**Review by ammolitor — CHANGES_REQUESTED** [review]
Please address the lingering devtoolset-8 package, otherwise, this looks good.

File: docker/centos7/Dockerfile (hunk 1 above)
Comment (sfc-gh-dluan) [reply docker/centos7/Dockerfile]: Done. I left it in during testing and accidentally forgot to remove it.

**Comment by ammolitor** [comment]
The `boringssl` build is failing tests, @sfc-gh-dluan can you dig in and see why? _Maybe_ the changes we are making at lines 274-276 are no longer necessary?

**Comment by sfc-gh-dluan** [comment]
I think there are some sort of an expiring certificate error with the `boringssl` tests - they recently published this to fix it. 
 https://boringssl.googlesource.com/boringssl/+/28f96c2686459add7acedcd97cb841030bdda019

**Pull request merged**